internal/zinc-core/src/main/scala/sbt/internal/inc/IncrementalCommon.scala

```diff
@@ -385,7 +386,7 @@ private[inc] abstract class IncrementalCommon(
         val changed0 = new java.util.HashSet[VirtualFileRef]
         val removed0 = new java.util.HashSet[VirtualFileRef]
         val unmodified0 = new java.util.HashSet[VirtualFileRef]
-        sources.foreach {
+        new ParVector(sources.toVector).foreach {
```

Comment:
👍

Reply:
I'm starting to see some under-detection around 30~50% of the times when I call `zincJVM2_13/testOnly sbt.inc.IncrementalCompilerSpec`

```
[info] - should not compile anything if source has not changed *** FAILED ***
[info]   result2.hasModified() was true (IncrementalCompilerSpec.scala:57)
```

Maybe the HashSet needs to be a concurrent-ready Set?

Reply:
I have a fix for this here - https://github.com/sbt/zinc/pull/849/commits/4f14d025260a9c78c09e1706262a79fe5cf6be07

Reply:
Good catch. I had noticed that ci was a little flaky and was wondering what could have been going on.